R/checkers.R

```diff
@@ -45,3 +56,344 @@ check_elements_non_neg <- function(x, arg = "x", call = rlang::caller_env()) {
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#'
+#' @return NULL, invisibly
+check_no_missingness <- function(x, arg = "x", call = rlang::caller_env()) {
+  is_missing <- rlang::are_na(x)
+
+  if (any(is_missing)) {
+    cli::cli_abort(
+      c("{.arg {arg}} has missing values",
+        "i" = "Missing values are not supported in {.arg {arg}}",
+        "!" = "Missing element(s) index: {.val {which(is_missing)}}"
+      ),
+      call = call,
+      class = "wwinference_input_data_error"
+    )
+  }
+  invisible()
+}
+
+#' Check that the vector of population sizes for the global catchment area
+#' has only a single value
+#'
+#' @description
+#' This function  checks that "global" population sizes in the data passed in as
+#' count data contain only a single value of the catchment areas population
+#' size. Multiple values might indicate either that there are multiple count
+#' data streams being passed in (which is not currently supported) or that
+#' there is a time varying population size (which is also not currently
+#' supported). This function is specific to the
+#' current version of the model, and will
+#' be deprecated for a more general `check_for_single_value()` once
+#' additional model functionality has been added.
+#'
+#'
+#' @param x the vector to check
+#' @param arg the name of the vector to check
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#'
+#' @return NULL, invisibly
+check_global_pop <- function(x, arg = "x", call = rlang::caller_env()) {
+  unique_pops <- unique(x)
+
+  if (length(unique_pops) > 1) {
+    cli::cli_abort(
+      c("{.arg {arg}} has more than one global population size",
+        "i" = c(
+          "Multiple/time-varying count catchment area populations",
+          "are not currently supported. Check that data is from a single",
+          "location, and if so, consider replacing with an average",
+          "population size over the inference period"
+        )
+      ),
+      call = call,
+      class = "wwinference_input_data_error"
+    )
+  }
+  invisible()
+}
+
+#' Check that there are no repeated elements in the vector of dates
+#' corresponding to count observations
+#'
+#' @description
+#' This function  checks that the dates in the data passed in as count data are
+#' not repeated, which might indicate multiple count data stream. This
+#' functionality is not currently supported. This function is specific to the
+#' current version of the model, and will
+#' be deprecated for a more general `check_for_repeat_elements()` once
+#' additional model functionality has been added.
+#'
+#'
+#' @param x the vector to check
+#' @param arg the name of the vector to check
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#'
+#' @return NULL, invisibly
+check_for_repeat_dates <- function(x, arg = "x", call = rlang::caller_env()) {
```

Comment:
this doesn't seem specific to dates in the implementation. Seems like it is best done with something like [`assertr`](https://docs.ropensci.org/assertr/articles/assertr.html#assertrs-predicates)'s uniqueness checking